### PR TITLE
Added Chat Functionality

### PR DESCRIPTION
**Pull request by TomFanella4**
## Changes
1. Users can view chats they are in
1. User can send and receive messages in real time

### CODE REVIEW

File: client/app/firebase/index.js [hunk 1]
```diff
@@ -69,3 +69,41 @@ export const modifyFirebaseUserProfile = async (key, value) => {
...
+    messagesRef = firebase.database().ref(`messages/${_id}`)
+    .endAt(null, endId).limitToLast(limit + 1)
+  :
+    messagesRef = firebase.database().ref(`messages/${_id}`).limitToLast(limit);
+
+  messagesRef.once('value', snapshot => {
+    const messages = Object.values(snapshot.val()).reverse();
+    endId && messages.shift();
+    updateFn(messages);
+    messagesRef.off();
```
Comment (mpace965) [inline client/app/firebase/index.js]: Why do you need to do `off` here? It looks like the point of [`once`](https://firebase.google.com/docs/reference/js/firebase.database.Query#once) is so you don't need to call `off`.

File: client/app/screens/ChatsScreen.js [hunk 2]
```diff
@@ -1,26 +1,90 @@
...
-export default class ChatsScreen extends React.Component {
+class ChatsScreen extends React.Component {
   static navigationOptions = {
     title: 'Chats',
   };
 
+  _viewChatThread(_id, name) {
+    this.props.navigation.navigate('ChatScreen', {
+      _id, name
+    });
```
Comment (mpace965) [inline client/app/screens/ChatsScreen.js]: nit: seems like this is short enough to be on one line

**Review by mpace965 — APPROVED** [review]
Really good work here. Had a small question and a nitpick

File: client/app/screens/ChatsScreen.js (hunk 2 above)
Comment (TomFanella4) [reply client/app/screens/ChatsScreen.js]: Reasonable

File: client/app/firebase/index.js (hunk 1 above)
Comment (TomFanella4) [reply client/app/firebase/index.js]: I forgot to remove it from the `.on()` implementation

**Pull request merged**